### DISCUSSION TITLE
Code scanning alerts found in minified code from webpack

### DISCUSSION THREAD

**Discussion by dastrobu** [github/codeql#15293] [Q&A]
After heaving set up CodeQL scanning it found some issue in the generated JavaScript files in the dist folder. 
The file is generated by webpack and, hence, not kept in the repo. 
That is why we cannot see where the issue is coming from:

![image](https://github.com/github/codeql/assets/1847260/595c6497-7084-4651-951c-01de259ac6fe)

Even if we had the file in the repository, it would be challenging to understand the problems in the minified JavaScript code. Additionally, it would be difficult to determine how to modify the original sources to address these issues.

My question is, how to handle these cases efficiently? 
I could of course add the `dist` folder to [`paths-ignore`](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/customizing-your-advanced-setup-for-code-scanning#specifying-directories-to-scan). However, I am hesitant to ignore the code that is actually executed in the end and only scan the code that is used as input for webpack.

**Comment by mariusp12** [comment]
We also have the same problem: what should we do with the issues in the generated JS of an Angular application.

**Comment by aeisenberg** [comment]
Our recommendation is to add `dist` to paths-ignore. The `dist` folder contains minified versions of your own code, which is already checked by codeql. It also contains library code, which is checked by dependabot. So, everything in the folder should already be covered.

**Reply by justinmchase** [reply]
how?

**Reply by mbg** [reply]
@justinmchase Could you clarify what you are asking about? If your question is about how to add `dist` to `paths-ignore`, the [documentation](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/customizing-your-advanced-setup-for-code-scanning#specifying-directories-to-scan) that was linked to in the original question should answer that.

**Reply by justinmchase** [reply]
Sorry, I don't have a codeql workflow, Im using the built in codeql tool and I'm wondering if there is a way to configure this without having to replace it with an entirely custom workflow.

**Reply by mbg** [reply]
Assuming that you are using [default setup](https://docs.github.com/en/code-security/code-scanning/enabling-code-scanning/configuring-default-setup-for-code-scanning#about-default-setup), then I don't believe you can customise it in this way. You would have to migrate to an advanced (custom) workflow.

**Reply by justinmchase** [reply]
Ok, thanks, I was coming to that conclusion as well. Our enterprise is managing the default settings and rules so I think I'm stuck.

Consider this a feature request to have exclusions in the default setup configuration.

**Reply by mbg** [reply]
Thanks for confirming @justinmchase -- it makes sense and we have noted the feature request.
